pyqtgraph/graphicsItems/BarGraphItem.py

```diff
@@ -58,93 +60,103 @@ def setOpts(self, **opts):
         self._shape = None
         self.update()
         self.informViewBoundsChanged()
-        
-    def drawPicture(self):
-        self.picture = QtGui.QPicture()
-        self._shape = QtGui.QPainterPath()
-        p = QtGui.QPainter(self.picture)
-        
-        pen = self.opts['pen']
-        pens = self.opts['pens']
-        
-        if pen is None and pens is None:
-            pen = getConfigOption('foreground')
-        
-        brush = self.opts['brush']
-        brushes = self.opts['brushes']
-        if brush is None and brushes is None:
-            brush = (128, 128, 128)
+
+    def setPen(self, *args, **kargs):
+        """Set the pen used to draw the curve."""
+        self.setOpts(pen=fn.mkPen(*args, **kargs))
+
+    def setBrush(self, *args, **kargs):
+        """Set the brush used when filling the area under the curve"""
+        self.setOpts(brush=fn.mkBrush(*args, **kargs))
+
+    def setData(self, **kargs):
         
         def asarray(x):
             if x is None or np.isscalar(x) or isinstance(x, np.ndarray):
                 return x
             return np.array(x)
 
+        x = asarray(kargs.pop('x', None))
+        self.x0 = asarray(kargs.pop('x0', None))
+        x1 = asarray(kargs.pop('x1', None))
+        self.width = asarray(kargs.pop('width', None))
```

Comment:
- setOpts (and self.opts) are used in pyqtgraph for options not data most of the times.
- documentation for BarGraphItem explicitly says that any combination of x, x0, x1, y, y0, y1, width and height is allowed, so I would not guess about what one typycally sets. 
- getData was broken before (it could return unset data), fixed in the last update.